rmw/include/rmw/rmw.h

```diff
@@ -448,10 +448,21 @@ rmw_publish_loaned_message(
  * Query the underlying middleware to determine how many subscriptions are
  * matched to a given publisher.
  *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | No
+ * Thread-Safe        | No
+ * Uses Atomics       | Maybe [1]
+ * Lock-Free          | Maybe [1]
+ * <i>[1] rmw implementation defined, check the implementation documentation</i>
+ *
  * \param[in] publisher the publisher object to inspect
  * \param[out] subscription_count the number of subscriptions matched
  * \return `RMW_RET_OK` if successful, or
  * \return `RMW_RET_INVALID_ARGUMENT` if either argument is null, or
+ * \return `RMW_RET_INCORRECT_RMW_IMPLEMENTATION` if node or publisher
```

Comment:
```suggestion
 * \return `RMW_RET_INCORRECT_RMW_IMPLEMENTATION` if publisher
```

Or maybe I'm wrong.  I guess my question is; why is this different than `rmw_subscription_count_matched_publishers`?

Reply:
This seems like the right change to me. It's not possible AFAIK for the node which is referenced from within the publisher to not match the publisher, so checking the publisher only is fine I think.

Reply:
You're spot on. A left over. See https://github.com/ros2/rmw/pull/262/commits/1083ea1c855f9d56de25ca8b307c6b3a2f7421aa.